### ISSUE TITLE
[Build] Cannot run react-explorer because drivelist.node is missing

### ISSUE THREAD

**Issue by saifis** [warpdesign/react-explorer#166]
Trying to test it out and tried to build it locally, 

`npx electron ./build/main.js
`
```
Uncaught (in promise) Error: Could not locate the bindings file. Tried:
 → /Users/xxxx/Documents/react/react-explorer/build/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/build/Debug/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/build/Release/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/out/Debug/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/Debug/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/out/Release/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/Release/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/build/default/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/compiled/10.2.0/darwin/x64/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/addon-build/release/install-root/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/addon-build/debug/install-root/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/addon-build/default/install-root/drivelist.node
 → /Users/xxxx/Documents/react/react-explorer/lib/binding/node-v64-darwin-x64/drivelist.node
```

And sure enough none of those files are in the build directory, I feel I'm doing something wrong that has nothing to do with the particular project and its something more general, but I can't seem to find good info on it.

for reference I'm building on a Mac

```
macOS Catalina 10.15.6
npm -v 6.14.4
node -v v14.4.0

```

**Comment by warpdesign** [comment]
I can reproduce the problem with a fresh check out of the respository: `drivelist` is a native node dependency that should be copied to the build directory. For some reason it is not.

Could you try typing:

> cp ./node_modules/drivelist/build/Release/drivelist.node ./build

This should fix the problem.

If that fails, could you list the content of the drivelist build directory?

> ls node_modules/drivelist/build/Release

**Comment by saifis** [comment]
Thanks for the response, it did change the errors, pasting the console output. thank you for the assistance since I feel pretty lost at the moment.

```
Download the React DevTools for a better development experience: https://fb.me/react-devtools
You might need to use a local HTTP server (instead of file://): https://fb.me/react-devtools-faq
logger.js:23 i18next: languageChanged en
logger.js:23 i18next: initialized Object
logger.js:23 i18next: languageChanged ja
settingsState.ts:85 setting language to ja
/Users/xxxx/Documents/react/react-explorer/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/renderer/security-warnings.js:184 Electron Security Warning (Insecure Content-Security-Policy) This renderer process has either no Content Security
      Policy set or a policy with "unsafe-eval" enabled. This exposes users of
      this app to unnecessary security risks.
 
For more information and help, consult
https://electronjs.org/docs/tutorial/security.
 This warning will not show up
once the app is packaged.
warnAboutInsecureCSP @ /Users/xxxx/Documents/react/react-explorer/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/renderer/security-warnings.js:184
loadHandler @ /Users/xxxx/Documents/react/react-explorer/node_modules/electron/dist/Electron.app/Contents/Resources/electron.asar/renderer/init.js:182
index.tsx:51 error preparing fake folders Error: Command failed: /Users/leo/tmp_ftp.sh
/bin/sh: /Users/leo/tmp_ftp.sh: No such file or directory


    at ChildProcess.exithandler (child_process.js:303)
    at ChildProcess.emit (events.js:182)
    at maybeClose (internal/child_process.js:961)
    at Socket.stream.socket.on (internal/child_process.js:380)
    at Socket.emit (events.js:182)
    at Pipe._handle.close [as _onclose] (net.js:596)
react-dom.development.js:88 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Provider
printWarning @ react-dom.development.js:88
warn @ react-dom.development.js:51
./node_modules/react-dom/cjs/react-dom.development.js.ReactStrictModeWarnings.flushPendingUnsafeLifecycleWarnings @ react-dom.development.js:11377
flushRenderPhaseStrictModeWarningsInDEV @ react-dom.development.js:23112
commitRootImpl @ react-dom.development.js:22396
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
commitRoot @ react-dom.development.js:22381
finishSyncRender @ react-dom.development.js:21807
performSyncWorkOnRoot @ react-dom.development.js:21793
scheduleUpdateOnFiber @ react-dom.development.js:21188
updateContainer @ react-dom.development.js:24373
(anonymous) @ react-dom.development.js:24758
unbatchedUpdates @ react-dom.development.js:21903
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
App.renderApp @ index.tsx:73
App.init @ index.tsx:66
App.tsx:75 App:constructor {}
winState.ts:24 WinState 0 1
appState.ts:90 adding view 0 /tmp/react-explorer
viewState.ts:18 viewState/addCache /tmp/react-explorer
Fs.ts:146 got FS {icon: "database", name: "local", description: "Local Filesystem", options: {…}, canread: ƒ, …}
fileState.ts:137 this.fs {icon: "database", name: "local", description: "Local Filesystem", options: {…}, canread: ƒ, …}
appState.ts:90 adding view 1 /tmp/react-explorer
viewState.ts:18 viewState/addCache /tmp/react-explorer
Fs.ts:146 got FS {icon: "database", name: "local", description: "Local Filesystem", options: {…}, canread: ƒ, …}
fileState.ts:137 this.fs {icon: "database", name: "local", description: "Local Filesystem", options: {…}, canread: ƒ, …}
2fileState.ts:382 fileState: cd /tmp/react-explorer /tmp/react-explorer
react-dom.development.js:88 Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move code with side effects to componentDidMount, and set initial state in the constructor.
* Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: App
printWarning @ react-dom.development.js:88
warn @ react-dom.development.js:51
./node_modules/react-dom/cjs/react-dom.development.js.ReactStrictModeWarnings.flushPendingUnsafeLifecycleWarnings @ react-dom.development.js:11371
flushRenderPhaseStrictModeWarningsInDEV @ react-dom.development.js:23112
commitRootImpl @ react-dom.development.js:22396
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
commitRoot @ react-dom.development.js:22381
finishSyncRender @ react-dom.development.js:21807
performSyncWorkOnRoot @ react-dom.development.js:21793
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
unbatchedUpdates @ react-dom.development.js:21909
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
App.renderApp @ index.tsx:73
App.init @ index.tsx:66
PrefsDialog.tsx:241 Uncaught TypeError: Cannot read property 'lang' of undefined
    at PrefsDialogClass../src/components/dialogs/PrefsDialog.tsx.PrefsDialogClass.render (PrefsDialog.tsx:241)
    at finishClassComponent (react-dom.development.js:17160)
    at updateClassComponent (react-dom.development.js:17110)
    at beginWork (react-dom.development.js:18620)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at beginWork$1 (react-dom.development.js:23203)
    at performUnitOfWork (react-dom.development.js:22157)
    at workLoopSync (react-dom.development.js:22130)
./src/components/dialogs/PrefsDialog.tsx.PrefsDialogClass.render @ PrefsDialog.tsx:241
finishClassComponent @ react-dom.development.js:17160
updateClassComponent @ react-dom.development.js:17110
beginWork @ react-dom.development.js:18620
callCallback @ react-dom.development.js:188
invokeGuardedCallbackDev @ react-dom.development.js:237
invokeGuardedCallback @ react-dom.development.js:292
beginWork$1 @ react-dom.development.js:23203
performUnitOfWork @ react-dom.development.js:22157
workLoopSync @ react-dom.development.js:22130
performSyncWorkOnRoot @ react-dom.development.js:21756
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
unbatchedUpdates @ react-dom.development.js:21909
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
App.renderApp @ index.tsx:73
App.init @ index.tsx:66
Nav.tsx:76 render nav false
LeftPanel.tsx:98 languageChanged
LeftPanel.tsx:77 checking for WSL
react-dom.development.js:88 Warning: componentWillMount has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move code with side effects to componentDidMount, and set initial state in the constructor.
* Rename componentWillMount to UNSAFE_componentWillMount to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: HotkeysTarget(LogUIClass), LeftPanelClass, NavComponent, SideViewClass
printWarning @ react-dom.development.js:88
warn @ react-dom.development.js:51
./node_modules/react-dom/cjs/react-dom.development.js.ReactStrictModeWarnings.flushPendingUnsafeLifecycleWarnings @ react-dom.development.js:11371
flushRenderPhaseStrictModeWarningsInDEV @ react-dom.development.js:23112
commitRootImpl @ react-dom.development.js:22396
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
commitRoot @ react-dom.development.js:22381
finishSyncRender @ react-dom.development.js:21807
performSyncWorkOnRoot @ react-dom.development.js:21793
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
unbatchedUpdates @ react-dom.development.js:21909
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
App.renderApp @ index.tsx:73
App.init @ index.tsx:66
react-dom.development.js:19527 The above error occurred in the <PrefsDialogClass> component:
    in PrefsDialogClass (created by inject-PrefsDialogClass-with-settingsState)
    in inject-PrefsDialogClass-with-settingsState (created by NamespacesConsumerComponent)
    in NamespacesConsumerComponent (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in WithMergedOptions(NamespacesConsumerComponent) (created by LoadNamespace)
    in LoadNamespace (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in LoadNamespace(inject-PrefsDialogClass-with-settingsState) (created by App)
    in Provider (created by App)
    in App (created by inject-App-with-settingsState)
    in inject-App-with-settingsState (created by NamespacesConsumerComponent)
    in NamespacesConsumerComponent (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in WithMergedOptions(NamespacesConsumerComponent) (created by LoadNamespace)
    in LoadNamespace (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in LoadNamespace(inject-App-with-settingsState)
    in Provider
    in I18nextProvider
    in Unknown

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
logCapturedError @ react-dom.development.js:19527
logError @ react-dom.development.js:19564
update.callback @ react-dom.development.js:20708
callCallback @ react-dom.development.js:12490
commitUpdateQueue @ react-dom.development.js:12511
commitLifeCycles @ react-dom.development.js:19883
commitLayoutEffects @ react-dom.development.js:22803
callCallback @ react-dom.development.js:188
invokeGuardedCallbackDev @ react-dom.development.js:237
invokeGuardedCallback @ react-dom.development.js:292
commitRootImpl @ react-dom.development.js:22541
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
commitRoot @ react-dom.development.js:22381
finishSyncRender @ react-dom.development.js:21807
performSyncWorkOnRoot @ react-dom.development.js:21793
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
unbatchedUpdates @ react-dom.development.js:21909
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
App.renderApp @ index.tsx:73
App.init @ index.tsx:66
fileState.ts:429 error cd/list for path /tmp/react-explorer error was Error: ENOENT: no such file or directory, lstat '/tmp/react-explorer'
    at fs.lstatSync (fs.js:936)
    at Object.fs.lstatSync (ELECTRON_ASAR.js:259)
    at LocalApi.<anonymous> (FsLocal.ts:190)
    at step (FsGeneric.ts:173)
    at Object.next (FsGeneric.ts:173)
    at FsGeneric.ts:173
    at new Promise (<anonymous>)
    at ./src/services/plugins/FsLocal.ts.__awaiter (FsGeneric.ts:173)
    at LocalApi../src/services/plugins/FsLocal.ts.LocalApi.isDir (FsLocal.ts:189)
    at LocalApi../src/services/plugins/FsLocal.ts.LocalApi.cd (FsLocal.ts:59)
react-dom.development.js:88 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in SideViewClass (created by inject-SideViewClass-with-appState)
    in inject-SideViewClass-with-appState (created by Context.Consumer)
    in DropTarget(inject-SideViewClass-with-appState) (created by NamespacesConsumerComponent)
    in NamespacesConsumerComponent (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in WithMergedOptions(NamespacesConsumerComponent) (created by LoadNamespace)
    in LoadNamespace (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in LoadNamespace(DropTarget(inject-SideViewClass-with-appState)) (created by App)
    in div (created by App)
    in Provider (created by App)
    in App (created by inject-App-with-settingsState)
    in inject-App-with-settingsState (created by NamespacesConsumerComponent)
    in NamespacesConsumerComponent (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in WithMergedOptions(NamespacesConsumerComponent) (created by LoadNamespace)
    in LoadNamespace (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in LoadNamespace(inject-App-with-settingsState)
    in Provider
    in I18nextProvider
    in Unknown
printWarning @ react-dom.development.js:88
error @ react-dom.development.js:60
warnAboutUpdateOnUnmountedFiberInDEV @ react-dom.development.js:23161
scheduleUpdateOnFiber @ react-dom.development.js:21169
enqueueForceUpdate @ react-dom.development.js:12678
./node_modules/react/cjs/react.development.js.Component.forceUpdate @ react.development.js:490
(anonymous) @ index.module.js:756
./node_modules/mobx/lib/mobx.module.js.Reaction.runReaction @ mobx.module.js:3305
runReactionsHelper @ mobx.module.js:3441
reactionScheduler @ mobx.module.js:3419
(anonymous) @ mobx.module.js:3448
batchedUpdates$1 @ react-dom.development.js:21856
reactionScheduler @ mobx.module.js:3448
runReactions @ mobx.module.js:3424
endBatch @ mobx.module.js:2895
endAction @ mobx.module.js:947
executeAction @ mobx.module.js:914
res @ mobx.module.js:902
(anonymous) @ fileState.ts:430
Promise.catch (async)
./src/state/fileState.ts.FileState.cwd @ fileState.ts:427
(anonymous) @ Fs.ts:173
step @ gui.js:97200
(anonymous) @ gui.js:97181
fulfilled @ gui.js:97172
Promise.then (async)
step @ gui.js:97174
(anonymous) @ gui.js:97175
./src/services/Fs.ts.__awaiter @ gui.js:97171
decorator @ Fs.ts:163
executeAction @ mobx.module.js:911
res @ mobx.module.js:902
./src/state/fileState.ts.FileState.cd @ fileState.ts:397
executeAction @ mobx.module.js:911
res @ mobx.module.js:902
./src/state/viewState.ts.ViewState.setVisibleCache @ viewState.ts:53
executeAction @ mobx.module.js:911
res @ mobx.module.js:902
./src/state/winState.ts.WinState.initState @ winState.ts:60
executeAction @ mobx.module.js:911
res @ mobx.module.js:902
./src/state/appState.ts.AppState.initViewState @ appState.ts:113
executeAction @ mobx.module.js:911
res @ mobx.module.js:902
AppState @ appState.ts:93
App @ App.tsx:94
constructClassInstance @ react-dom.development.js:12880
updateClassComponent @ react-dom.development.js:17100
beginWork @ react-dom.development.js:18620
beginWork$1 @ react-dom.development.js:23179
performUnitOfWork @ react-dom.development.js:22157
workLoopSync @ react-dom.development.js:22130
performSyncWorkOnRoot @ react-dom.development.js:21756
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
unbatchedUpdates @ react-dom.development.js:21909
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
App.renderApp @ index.tsx:73
App.init @ index.tsx:66
fileState.ts:429 error cd/list for path /tmp/react-explorer error was Error: ENOENT: no such file or directory, lstat '/tmp/react-explorer'
    at fs.lstatSync (fs.js:936)
    at Object.fs.lstatSync (ELECTRON_ASAR.js:259)
    at LocalApi.<anonymous> (FsLocal.ts:190)
    at step (FsGeneric.ts:173)
    at Object.next (FsGeneric.ts:173)
    at FsGeneric.ts:173
    at new Promise (<anonymous>)
    at ./src/services/plugins/FsLocal.ts.__awaiter (FsGeneric.ts:173)
    at LocalApi../src/services/plugins/FsLocal.ts.LocalApi.isDir (FsLocal.ts:189)
    at LocalApi../src/services/plugins/FsLocal.ts.LocalApi.cd (FsLocal.ts:59)
LeftPanel.tsx:125 places updated: need to rebuild nodes
react-dom.development.js:88 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in LeftPanelClass (created by inject-LeftPanelClass-with-appState)
    in inject-LeftPanelClass-with-appState (created by NamespacesConsumerComponent)
    in NamespacesConsumerComponent (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in WithMergedOptions(NamespacesConsumerComponent) (created by LoadNamespace)
    in LoadNamespace (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in LoadNamespace(inject-LeftPanelClass-with-appState) (created by App)
    in div (created by App)
    in Provider (created by App)
    in App (created by inject-App-with-settingsState)
    in inject-App-with-settingsState (created by NamespacesConsumerComponent)
    in NamespacesConsumerComponent (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in WithMergedOptions(NamespacesConsumerComponent) (created by LoadNamespace)
    in LoadNamespace (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in LoadNamespace(inject-App-with-settingsState)
    in Provider
    in I18nextProvider
    in Unknown
printWarning @ react-dom.development.js:88
error @ react-dom.development.js:60
warnAboutUpdateOnUnmountedFiberInDEV @ react-dom.development.js:23161
scheduleUpdateOnFiber @ react-dom.development.js:21169
enqueueSetState @ react-dom.development.js:12639
./node_modules/react/cjs/react.development.js.Component.setState @ react.development.js:471
./src/components/LeftPanel.tsx.LeftPanelClass.buildNodes @ LeftPanel.tsx:268
(anonymous) @ LeftPanel.tsx:126
executeAction @ mobx.module.js:911
res @ mobx.module.js:902
reactionRunner @ mobx.module.js:1456
(anonymous) @ mobx.module.js:1434
./node_modules/mobx/lib/mobx.module.js.Reaction.runReaction @ mobx.module.js:3305
runReactionsHelper @ mobx.module.js:3441
reactionScheduler @ mobx.module.js:3419
(anonymous) @ mobx.module.js:3448
batchedUpdates$1 @ react-dom.development.js:21856
reactionScheduler @ mobx.module.js:3448
runReactions @ mobx.module.js:3424
endBatch @ mobx.module.js:2895
./node_modules/mobx/lib/mobx.module.js.BaseAtom.reportChanged @ mobx.module.js:64
./node_modules/mobx/lib/mobx.module.js.ObservableArrayAdministration.notifyArraySplice @ mobx.module.js:434
./node_modules/mobx/lib/mobx.module.js.ObservableArrayAdministration.spliceWithArray @ mobx.module.js:382
./node_modules/mobx/lib/mobx.module.js.ObservableArray.replace @ mobx.module.js:480
./src/state/favoritesState.ts.FavoritesState.buildPlaces @ favoritesState.ts:53
(anonymous) @ favoritesState.ts:97
step @ appState.ts:50
(anonymous) @ appState.ts:50
fulfilled @ appState.ts:50
Promise.then (async)
step @ appState.ts:50
(anonymous) @ appState.ts:50
./src/state/favoritesState.ts.__awaiter @ appState.ts:50
FavoritesState.checkForNewDrives @ favoritesState.ts:93
./src/state/favoritesState.ts.FavoritesState.launchTimeout @ favoritesState.ts:67
(anonymous) @ favoritesState.ts:76
step @ appState.ts:50
(anonymous) @ appState.ts:50
(anonymous) @ appState.ts:50
./src/state/favoritesState.ts.__awaiter @ appState.ts:50
./src/state/favoritesState.ts.FavoritesState.buildDrivesList @ favoritesState.ts:74
FavoritesState @ favoritesState.ts:121
AppState @ appState.ts:55
App @ App.tsx:94
constructClassInstance @ react-dom.development.js:12880
updateClassComponent @ react-dom.development.js:17100
beginWork @ react-dom.development.js:18620
beginWork$1 @ react-dom.development.js:23179
performUnitOfWork @ react-dom.development.js:22157
workLoopSync @ react-dom.development.js:22130
performSyncWorkOnRoot @ react-dom.development.js:21756
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
unbatchedUpdates @ react-dom.development.js:21909
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
App.renderApp @ index.tsx:73
App.init @ index.tsx:66
PrefsDialog.tsx:241 Uncaught (in promise) TypeError: Cannot read property 'lang' of undefined
    at PrefsDialogClass../src/components/dialogs/PrefsDialog.tsx.PrefsDialogClass.render (PrefsDialog.tsx:241)
    at finishClassComponent (react-dom.development.js:17160)
    at updateClassComponent (react-dom.development.js:17110)
    at beginWork (react-dom.development.js:18620)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at beginWork$1 (react-dom.development.js:23203)
    at performUnitOfWork (react-dom.development.js:22157)
    at workLoopSync (react-dom.development.js:22130)
./src/components/dialogs/PrefsDialog.tsx.PrefsDialogClass.render @ PrefsDialog.tsx:241
finishClassComponent @ react-dom.development.js:17160
updateClassComponent @ react-dom.development.js:17110
beginWork @ react-dom.development.js:18620
callCallback @ react-dom.development.js:188
invokeGuardedCallbackDev @ react-dom.development.js:237
invokeGuardedCallback @ react-dom.development.js:292
beginWork$1 @ react-dom.development.js:23203
performUnitOfWork @ react-dom.development.js:22157
workLoopSync @ react-dom.development.js:22130
performSyncWorkOnRoot @ react-dom.development.js:21756
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
unbatchedUpdates @ react-dom.development.js:21909
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
App.renderApp @ index.tsx:73
App.init @ index.tsx:66
2fs.js:142 Uncaught (in promise) Error: No such file or directory
    at fs.lstatSync (fs.js:936)
    at Object.fs.lstatSync (ELECTRON_ASAR.js:259)
    at LocalApi.<anonymous> (FsLocal.ts:190)
    at step (FsGeneric.ts:173)
    at Object.next (FsGeneric.ts:173)
    at FsGeneric.ts:173
    at new Promise (<anonymous>)
    at ./src/services/plugins/FsLocal.ts.__awaiter (FsGeneric.ts:173)
    at LocalApi../src/services/plugins/FsLocal.ts.LocalApi.isDir (FsLocal.ts:189)
    at LocalApi../src/services/plugins/FsLocal.ts.LocalApi.cd (FsLocal.ts:59)
handleErrorFromBinding @ fs.js:142
fs.lstatSync @ fs.js:936
fs.lstatSync @ ELECTRON_ASAR.js:259
(anonymous) @ FsLocal.ts:190
step @ FsGeneric.ts:173
(anonymous) @ FsGeneric.ts:173
(anonymous) @ FsGeneric.ts:173
./src/services/plugins/FsLocal.ts.__awaiter @ FsGeneric.ts:173
./src/services/plugins/FsLocal.ts.LocalApi.isDir @ FsLocal.ts:189
./src/services/plugins/FsLocal.ts.LocalApi.cd @ FsLocal.ts:59
./src/state/fileState.ts.FileState.cwd @ fileState.ts:407
(anonymous) @ Fs.ts:173
step @ gui.js:97200
(anonymous) @ gui.js:97181
fulfilled @ gui.js:97172
react-dom.development.js:88 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.
    in HotkeysTarget(LogUIClass) (created by NamespacesConsumerComponent)
    in NamespacesConsumerComponent (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in WithMergedOptions(NamespacesConsumerComponent) (created by LoadNamespace)
    in LoadNamespace (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in LoadNamespace(HotkeysTarget(LogUIClass)) (created by App)
    in Provider (created by App)
    in App (created by inject-App-with-settingsState)
    in inject-App-with-settingsState (created by NamespacesConsumerComponent)
    in NamespacesConsumerComponent (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in WithMergedOptions(NamespacesConsumerComponent) (created by LoadNamespace)
    in LoadNamespace (created by WithMergedOptions)
    in WithMergedOptions (created by Context.Consumer)
    in LoadNamespace(inject-App-with-settingsState)
    in Provider
    in I18nextProvider
    in Unknown
printWarning @ react-dom.development.js:88
error @ react-dom.development.js:60
warnAboutUpdateOnUnmountedFiberInDEV @ react-dom.development.js:23161
scheduleUpdateOnFiber @ react-dom.development.js:21169
enqueueForceUpdate @ react-dom.development.js:12678
./node_modules/react/cjs/react.development.js.Component.forceUpdate @ react.development.js:490
(anonymous) @ index.module.js:756
./node_modules/mobx/lib/mobx.module.js.Reaction.runReaction @ mobx.module.js:3305
runReactionsHelper @ mobx.module.js:3441
reactionScheduler @ mobx.module.js:3419
(anonymous) @ mobx.module.js:3448
batchedUpdates$1 @ react-dom.development.js:21856
reactionScheduler @ mobx.module.js:3448
runReactions @ mobx.module.js:3424
endBatch @ mobx.module.js:2895
endAction @ mobx.module.js:947
executeAction @ mobx.module.js:914
runInAction @ mobx.module.js:1157
(anonymous) @ Log.tsx:62
setTimeout (async)
pushLog @ Log.tsx:60
success @ Log.tsx:48
App @ App.tsx:105
constructClassInstance @ react-dom.development.js:12880
updateClassComponent @ react-dom.development.js:17100
beginWork @ react-dom.development.js:18620
beginWork$1 @ react-dom.development.js:23179
performUnitOfWork @ react-dom.development.js:22157
workLoopSync @ react-dom.development.js:22130
performSyncWorkOnRoot @ react-dom.development.js:21756
(anonymous) @ react-dom.development.js:11089
unstable_runWithPriority @ scheduler.development.js:653
runWithPriority$1 @ react-dom.development.js:11039
flushSyncCallbackQueueImpl @ react-dom.development.js:11084
flushSyncCallbackQueue @ react-dom.development.js:11072
unbatchedUpdates @ react-dom.development.js:21909
legacyRenderSubtreeIntoContainer @ react-dom.development.js:24757
render @ react-dom.development.js:24840
App.renderApp @ index.tsx:73
App.init @ index.tsx:66
```

**Comment by saifis** [comment]
I think I know what one of the errors are, my primary language was set to Japanese, and it seemed to be ending up with unidentified language object. Tried turning system language on mac to English but to no avail.

**Comment by saifis** [comment]
yep adding ja.json fixed it up, up and running now, thank you!

**Comment by warpdesign** [comment]
Oh, that's a bad bug you just found: this should not crash and should default to English.

I opened a new [issue](https://github.com/warpdesign/react-explorer/issues/167) for this specific problem. I'll fix it quickly.

Any chance you could make a PR with the Japanese translation of react-explorer?

**Comment by warpdesign** [comment]
@saifis I updated the issue's title to better reflect the first bug reported.

I have made a fix for the language bug #167: a new release should be ready by tomorrow.

**Comment by saifis** [comment]
Thanks, Imma find the time to make the Japanese locale file, now you really will have a ja.json file.
